packages/syft/src/syft/service/log/log_service.py

```diff
@@ -57,7 +70,18 @@ def get(
         result = self.stash.get_by_uid(context.credentials, uid)
         if result.is_err():
             return SyftError(message=str(result.err()))
-        return result.ok()
+
+        return Ok(result.ok().stdout)
+
+    @service_method(path="log.get_error", name="get_error", roles=ADMIN_ROLE_LEVEL)
+    def get_error(
+        self, context: AuthedServiceContext, uid: UID
+    ) -> Union[SyftSuccess, SyftError]:
+        result = self.stash.get_by_uid(context.credentials, uid)
+        if result.is_err():
+            return SyftError(message=str(result.err()))
+
+        return Ok(result.ok().stderr)
```

Comment:
Solves: `make sure that we add the error to the job, but in such a way that admins can see the error and users cannot.`

Reply:
The `Job.logs()` method has been updated to include an optional `stderr` boolean parameter, which is set to false by default. When this parameter is set to true, the method will invoke a specialized logging service designed to fetch stderr messages. However, since this service requires  `ADMIN_ROLE_LEVEL` privileges, common users will not be able to retrieve it.
<img width="996" alt="exception_logs" src="https://github.com/OpenMined/PySyft/assets/26658472/89cc6eed-db2e-482f-a70e-54c31a3c0c03">